### PR TITLE
Disable interrupts when inside mtx_lock or mtx_unlock

### PR DESCRIPTION
**Pull request by xairy**

### CODE REVIEW

File: mm/ktsan/ktsan.c [hunk 1]
```diff
@@ -329,7 +348,15 @@ void ktsan_memblock_free(void *addr, unsigned long size)
 void ktsan_mtx_pre_lock(void *addr, bool write, bool try)
 {
 	ENTER(false, true);
-	kt_mtx_pre_lock(thr, pc, (uptr_t)addr, write, try);
+
+	if (kt_thr_event_disable(thr, pc)) {
+		thr->irq_flags_before_mtx = kt_flags;
+		/* Set all disabled in kt_flags. */
+		kt_flags = arch_local_irq_save();
```
Comment (dvyukov) [inline mm/ktsan/ktsan.c]: PLease add a comment as to why we disable interrupts here.

File: mm/ktsan/ktsan.c [hunk 2]
```diff
@@ -869,7 +918,7 @@ EXPORT_SYMBOL(ktsan_write_range);
 
 void ktsan_func_entry(void *call_pc)
 {
-	ENTER(false, false);
+	ENTER(false, true);
```
Comment (dvyukov) [inline mm/ktsan/ktsan.c]: revert this

File: mm/ktsan/ktsan.c [hunk 3]
```diff
@@ -877,7 +926,7 @@ EXPORT_SYMBOL(ktsan_func_entry);
 
 void ktsan_func_exit(void)
 {
-	ENTER(false, false);
+	ENTER(false, true);
```
Comment (dvyukov) [inline mm/ktsan/ktsan.c]: revert this

**Comment by dvyukov** [comment]
LGTM with nits

File: mm/ktsan/ktsan.c (hunk 1 above)
Comment (xairy) [inline mm/ktsan/ktsan.c]: Done.

File: mm/ktsan/ktsan.c (hunk 2 above)
Comment (xairy) [inline mm/ktsan/ktsan.c]: Done.

File: mm/ktsan/ktsan.c (hunk 3 above)
Comment (xairy) [inline mm/ktsan/ktsan.c]: Done.

**Pull request merged**